src/frontend/src/components/modals/LicenseModal.tsx

```diff
@@ -0,0 +1,97 @@
+import { Trans, t } from '@lingui/macro';
+import {
+  Accordion,
+  Alert,
+  Divider,
+  Group,
+  LoadingOverlay,
+  Space,
+  Stack,
+  Tabs,
+  Text
+} from '@mantine/core';
+import { ContextModalProps } from '@mantine/modals';
+import { useQuery } from '@tanstack/react-query';
+
+import { api } from '../../App';
+import { ApiEndpoints } from '../../enums/ApiEndpoints';
+import { apiUrl } from '../../states/ApiState';
+import { StylishText } from '../items/StylishText';
+
+export function LicenceView(entries: any[]) {
+  return (
+    <Stack spacing="xs">
+      <Divider />
+      {entries?.length && (
+        <Accordion variant="contained" defaultValue="-">
+          {entries?.map((entry: any, index: number) => (
+            <Accordion.Item key={index} value={`entry-${index}`}>
+              <Accordion.Control>
+                <Group position="apart" grow>
+                  <Text>{entry.name}</Text>
+                  <Text>{entry.license}</Text>
+                  <Space />
+                  <Text>{entry.version}</Text>
+                </Group>
+              </Accordion.Control>
+              <Accordion.Panel>
+                {entry.licensetext || t`No license text available`}
+              </Accordion.Panel>
+            </Accordion.Item>
+          ))}
+        </Accordion>
+      )}
+    </Stack>
+  );
+}
+
+export function LicenseModal({}: ContextModalProps<{
+  modalBody: string;
+}>) {
+  const { data, isFetching, isError } = useQuery({
+    queryKey: ['license'],
+    queryFn: () =>
+      api
+        .get(apiUrl(ApiEndpoints.license))
+        .then((res) => res.data ?? {})
+        .catch(() => {})
+  });
+
+  return (
+    <Stack spacing="xs">
+      <Divider />
+      <LoadingOverlay visible={isFetching} />
+      {isFetching && (
+        <Text>
+          <Trans>Loading license information</Trans>
+        </Text>
+      )}
+      {isError ? (
+        <Alert color="red" title={t`Error`}>
+          <Text>
+            <Trans>Failed to fetch license information</Trans>
+          </Text>
+        </Alert>
+      ) : (
+        <Tabs defaultValue="backend">
+          <Tabs.List>
+            <Tabs.Tab value="backend">
+              <Trans>Backend Packages</Trans>
+            </Tabs.Tab>
+            <Tabs.Tab value="frontend">
+              <Trans>Frontend Packages</Trans>
+            </Tabs.Tab>
+          </Tabs.List>
+
+          <Tabs.Panel value="backend">
+            {LicenceView(data?.backend ?? [])}
```

Comment:
Is there a reason why we don't use a normal component here, but a function that returns jsx?

```suggestion
            <LicenceView data={data?.backend ?? []} />
```

And then:

```tsx
type LicenseType = {
  name: string;
  license: string;
  version: string;
  licensetext: string;
}

export function LicenceView({ data }: { data: LicenseType[] }) {
// ...
    {entries?.map((entry, index) => (
```

Reply:
No idea, I have not coded this but I will not type the response - see above comment

Reply:
I know that you potentially want to monkey patch this, but this has no effect on it as far as I understand you correct. This just types the expected format of one package's license. Providing some different format through patching the backend source will fail on the frontend.

But nevertheless, typing was just the cherry on top I tried to provide, while suggesting using a real react component  instead of a function that returns jsx to stick with the react syntax.